src/Filters/filt_order.jl

```diff
@@ -0,0 +1,458 @@
+#====================================================================
+
+Filter order estimation routines.
+Jordan R. Smith, 2021.
+
+Filter transformations translated from Scipy to Julia.
+
+SCIPY license:
+Copyright (c) 2001, 2002 Enthought, Inc.
+All rights reserved.
+
+Copyright (c) 2003-2017 SciPy Developers.
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+  a. Redistributions of source code must retain the above copyright notice,
+     this list of conditions and the following disclaimer.
+  b. Redistributions in binary form must reproduce the above copyright
+     notice, this list of conditions and the following disclaimer in the
+     documentation and/or other materials provided with the distribution.
+  c. Neither the name of Enthought nor the names of the SciPy Developers
+     may be used to endorse or promote products derived from this software
+     without specific prior written permission.
+
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS
+BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+THE POSSIBILITY OF SUCH DAMAGE.
+
+--- end of scipy license
```

Comment:
Is this compatible with our MIT license? I guess so, but don't knoiw.

Reply:
Yes, if your code is based on the SciPy, I don't argue against including this license, but rather wonder if we may include the code here and publish it under the project's MIT license. (And yes, if that is a problem, then we have to worry about remez_fir, too.) There's probably a site somewhere that collects this license compatibility stuff, but from a quick search, I couldn't find anything applicable.

Reply:
From what I've gathered, the only outstanding stipulation is the third clause of the BSD license, indicating the need for explicit written permission if the Julia code is derived from SciPy. The only portion that I had looked to SciPy for, (outside of the references listed in the `filter_order.jl` file), was how to optimize the passband edges in the bandpass/bandstop filter cases. Let me know if there's anything else I can do or provide @martinholters !